python/semantic_kernel/connectors/ai/function_choice_behavior.py

```diff
@@ -0,0 +1,244 @@
+            )
+        if isinstance(behavior, (RequiredFunction)):
+            return cls.Required(
+                auto_invoke=behavior.auto_invoke_kernel_functions,
+                function_fully_qualified_names=[behavior.function_fully_qualified_name]
+                if hasattr(behavior, "function_fully_qualified_name")
+                else None,
+            )
+        return cls(
+            enable_kernel_functions=behavior.enable_kernel_functions,
+            maximum_auto_invoke_attempts=behavior.max_auto_invoke_attempts,
+        )
+
+    @property
+    def auto_invoke_kernel_functions(self):
+        """Return True if auto_invoke_kernel_functions is enabled."""
+        return self.maximum_auto_invoke_attempts > 0
+
+    @auto_invoke_kernel_functions.setter
+    def auto_invoke_kernel_functions(self, value: bool):
+        """Set the auto_invoke_kernel_functions property."""
+        self.maximum_auto_invoke_attempts = DEFAULT_MAX_AUTO_INVOKE_ATTEMPTS if value else 0
+
+    def _check_and_get_config(
+        self, kernel: "Kernel", fully_qualified_names: list[str] | None = None, filters: dict[str, Any] | None = {}
+    ) -> FunctionCallChoiceConfiguration:
+        """Check for missing functions and get the function call choice configuration."""
+        full_kernel_function_metadata = set(kernel.get_full_list_of_function_metadata())
+
+        if fully_qualified_names:
+            _check_for_missing_functions(fully_qualified_names, full_kernel_function_metadata)
+            return FunctionCallChoiceConfiguration(
+                available_functions=kernel.get_list_of_function_metadata({"included_functions": fully_qualified_names})
+            )
+        if filters:
+            return FunctionCallChoiceConfiguration(available_functions=kernel.get_list_of_function_metadata(filters))
+        return FunctionCallChoiceConfiguration(available_functions=kernel.get_full_list_of_function_metadata())
+
+    def configure(
+        self,
+        kernel: "Kernel",
+        update_settings_callback: Callable[..., None],
+        settings: "PromptExecutionSettings",
+    ) -> None:
+        """Configure the function choice behavior."""
+        if not self.enable_kernel_functions:
+            return
+
+        config = self.get_config(kernel)
+
+        if config:
+            update_settings_callback(config, settings, self.type)
+
+    def get_config(self, kernel: "Kernel") -> FunctionCallChoiceConfiguration:
+        """Get the function call choice configuration based on the type."""
+        return self._check_and_get_config(kernel, self.function_fully_qualified_names, self.filters)
+
+    @classmethod
+    def Auto(
+        cls,
+        auto_invoke: bool = True,
+        *,
+        filters: dict[
+            Literal["excluded_plugins", "included_plugins", "excluded_functions", "included_functions"], list[str]
+        ]
+        | None = None,
+        function_fully_qualified_names: list[str] | None = None,
```

Comment:
is this not a special filter case: included_functions = this list?

Reply:
It is. Per our chat, going to remove `function_fully_qualified_names` so we only deal with `filters`.